framework/src/org/apache/cordova/engine/SystemWebViewEngine.java

```diff
@@ -199,7 +199,19 @@ private void initWebViewSettings() {
         
         // Fix for CB-1405
         // Google issue 4641
-        settings.getUserAgentString();
+        String defaultUserAgent = settings.getUserAgentString();
+
+        // Fix for CB-3360
+        String overrideUserAgent = getCordovaWebView().getPreferences().getString("OverrideUserAgent", null);
```

Comment:
I'd like to avoid circular dependency if possible on CordovaWebView.

Rather than depending on CordovaWebView here, could you save the instance of CordovaPreferences passed to the constructor and use that instead? Would also need to add CordovaPreferences as a 2nd arg to the other constructor as well.